core/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/TransportPutMappingAction.java

```diff
@@ -67,25 +68,30 @@ protected ClusterBlockException checkBlock(PutMappingRequest request, ClusterSta
 
     @Override
     protected void masterOperation(final PutMappingRequest request, final ClusterState state, final ActionListener<PutMappingResponse> listener) {
-        final String[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, request);
-        PutMappingClusterStateUpdateRequest updateRequest = new PutMappingClusterStateUpdateRequest()
-                .ackTimeout(request.timeout()).masterNodeTimeout(request.masterNodeTimeout())
-                .indices(concreteIndices).type(request.type())
-                .updateAllTypes(request.updateAllTypes())
-                .source(request.source());
+        try {
+            final String[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, request);
+            PutMappingClusterStateUpdateRequest updateRequest = new PutMappingClusterStateUpdateRequest()
+                    .ackTimeout(request.timeout()).masterNodeTimeout(request.masterNodeTimeout())
+                    .indices(concreteIndices).type(request.type())
+                    .updateAllTypes(request.updateAllTypes())
+                    .source(request.source());
 
-        metaDataMappingService.putMapping(updateRequest, new ActionListener<ClusterStateUpdateResponse>() {
+            metaDataMappingService.putMapping(updateRequest, new ActionListener<ClusterStateUpdateResponse>() {
 
-            @Override
-            public void onResponse(ClusterStateUpdateResponse response) {
-                listener.onResponse(new PutMappingResponse(response.isAcknowledged()));
-            }
+                @Override
+                public void onResponse(ClusterStateUpdateResponse response) {
+                    listener.onResponse(new PutMappingResponse(response.isAcknowledged()));
+                }
 
-            @Override
-            public void onFailure(Throwable t) {
-                logger.debug("failed to put mappings on indices [{}], type [{}]", t, concreteIndices, request.type());
-                listener.onFailure(t);
-            }
-        });
+                @Override
+                public void onFailure(Throwable t) {
+                    logger.debug("failed to put mappings on indices [{}], type [{}]", t, concreteIndices, request.type());
+                    listener.onFailure(t);
+                }
+            });
+        } catch (IndexNotFoundException ex) {
```

Comment:
heh yeah I mean we can catch exception but I really wonder if we should do this on the caller of the method? I mean this can be really critical if we miss anything here?